challenges/10exercises.js

```diff
@@ -1,26 +1,181 @@
+
+// 5. Define a function sum() and a function multiply() that sums and multiplies (respectively) all the numbers in an array of numbers.
 // For example, sum([1,2,3,4]) should return 10, and multiply([1,2,3,4]) should return 24.
 
+  function sum(numbers) {
+    var total = 0;
+    for (var i = 0; i < numbers.length; i++) {
+      total = total + numbers[i];
+    }
+    return total;
+  };
+
+
+  function multiply(numbers) {
+
+      var total = 1;
+      for (var i = 0; i < numbers.length; i++) {
+          total = total * numbers[i];
+      }
+      return total;
+  }
+
+
 
-// Define a function reverse() that computes the reversal of a string.
+// 6. Define a function reverse() that computes the reversal of a string.
 // For example, reverse("jag testar") should return the string "ratset gaj".
 
-// Represent a small bilingual lexicon as a Javascript object in the following fashion
-//{"merry":"god", "christmas":"jul", "and":"och", "happy":gott", "new":"nytt", "year":"år"}
+  function reverse(text) {
+    return text.split('').reverse().join('');
+    // var textArray = text.split('');
+    // var reverseArray = textArray.reverse();
+    // var reversedText = textArray.join('');
+    // return reversedText;
+  }
+
+
+// 7. Represent a small bilingual lexicon as a Javascript object in the following fashion
+//{"merry":"god", "christmas":"jul", "and":"och", "happy":"gott", "new":"nytt", "year":"år"}
 //and use it to translate your Christmas cards from English into Swedish.
 
-//Write a function findLongestWord() that takes an array of words and returns the length of the longest one.
+var lexicon = {
+"merry":"god",
+"christmas":"jul",
+"and":"och",
+"happy":"gott",
+"new":"nytt",
+"year":"år"
+}
+
+function translate(text) {
+  return text.split(" ").reduce( function(accumulator, word) {
+    if (lexicon[word]) {
+      accumulator.push(lexicon[word])
+    }
+    return accumulator
+  }, []).join(" ")
+}
+
+//Other ways of doing it:
+// var textArray = text.split(" "); //['merry', 'christmas']
+// var translatedText = [];
+//
+// textArray.forEach( function(word) {
+//    if (lexicon[word]) {
+//      translatedText.push(lexicon[word])
+//    }
+// })
+//
+// return translatedText.join(" ");
+
+// for ( var i = 0; i < textArray.length; i++ ) {
+//  var currentWord = textArray[i];
+//  for ( var english in lexicon ) {
+//    if ( currentWord === english ) {
+//      translatedText.push(lexicon[english]);
+//    }
+//  }
+// }
+//when word in 'text' string equals to key in 'christmasCard' object, return value of that key
+
+
+
+//8. Write a function findLongestWord() that takes an array of words and returns the length of the longest one.
+
+function findLongestWord(text){
+  var textArray = text.split(" ");
+  var longestWord = 0;
+  for (var i=0; i < textArray.length; i++) {
+    if (textArray[i].length > longestWord) {
+      longestWord = textArray[i].length;
+    }
+  }
+return longestWord; //what if we want it to return the actual word?
+}
```

Comment:
Here is a solution that could easily return the word or the length of the longest word  by simply removing or adding the length operation on the end.
```
+function findLongestWord( arr ) {
 +  return arr.sort( ( previous, next ) => previous.length > next.length ).pop().length
 +}